subprojects/core/src/main/java/org/gradle/api/internal/tasks/userinput/UserInputHandler.java

```diff
@@ -20,44 +20,94 @@
 
 import javax.annotation.Nullable;
 import java.util.Collection;
+import java.util.function.Function;
 
 public interface UserInputHandler {
     /**
-     * Prompts the user with a yes/no question and returns the answer.
+     * Prompts the user with a yes/no question and returns the answer. Requires that the user explicitly type "yes" or "no".
      *
      * @param question The text of the question.
-     * @return the answer or {@code null} if not connected to a user console.
+     * @return the answer or {@code null} if not connected to a console.
      */
     @Nullable
     @UsedByScanPlugin
     Boolean askYesNoQuestion(String question);
 
     /**
-     * Prompts the user with a yes/no question and returns the answer.
+     * Prompts the user with a boolean question and returns the answer.
      *
      * @param question The text of the question.
-     * @return the answer or the given default if not connected to a user console.
+     * @param defaultValue The option to present to the user as the default choice, and the value to use when not connected to a console
+     * @return the answer or the given default if not connected to a console.
      */
-    boolean askYesNoQuestion(String question, boolean defaultValue);
+    boolean askBooleanQuestion(String question, boolean defaultValue);
```

Comment:
🤔 Since the semantics changed, it makes sense to rename this method, but I think the naming can be improved.

From the user interaction POV, it's still a yes/no question, but "boolean" name does not imply this. Also, having an `askIntQuestion` creates an impression that this question is actually true/false question, which it is not.

What about `askLenientYesNoQuestion()` or something similar?

Reply:
I'm not sure. I don't really want the name to imply that the user has to type 'yes' or 'no'. The client may be a UI with a checkbox, for example.

The intent of other method is quite different, as it requires the user to explicitly type 'yes' to agree to something (eg license terms, etc), so I don't want to use a similar name that implies similar semantics.